lib/bigcommerce/prometheus/integrations/railtie.rb

```diff
@@ -23,7 +23,7 @@ module Integrations
       #
       class Railtie < ::Rails::Railtie
         initializer 'zzz.bc_prometheus_ruby.configure_rails_initialization' do |app|
-          Bigcommerce::Prometheus::Instrumentors::Web.new(app: app).start
+          Bigcommerce::Prometheus::Instrumentors::Web.new(app: app).start unless ::Bigcommerce::Prometheus.web_disable_automatic_instrumentation
```

Comment:
I would do this within `Configuration` inside `self.extended`:

```
def self.extended(base)
  if railtie_enabled?
    Bigcommerce::Prometheus::Integrations::Railtie.config.before_initialize { base.reset }
  else
    base.reset
  end
end

def railtie_enabled?
  defined?(Rails) && !railtie_disabled
end
```

Reply:
I'm having trouble with this approach:

```
Unable to load application: NoMethodError: undefined method `railtie_enabled?' for Bigcommerce::Prometheus::Configuration:Module
.../bc-prometheus-ruby/lib/bigcommerce/prometheus/configuration.rb:68:in `extended': undefined method `railtie_enabled?' for Bigcommerce::Prometheus::Configuration:Module (NoMethodError)
```

I've gotta run now, so I'll pick up here first thing tomorrow morning.

Reply:
Nevermind, I'm steering you in the wrong direction. What I thought I was doing was preventing the Railtie from being loaded at all, but this would only change how we reset the configuration when the Railtie is loaded (totally different thing). The Railtie is actually loaded here https://github.com/bigcommerce/bc-prometheus-ruby/blob/main/lib/bigcommerce/prometheus.rb#L41 so to do what I was expected, we'd have to embed the logic in the Railtie itself, as you have it now.

Reply:
> so to do what we need, we have to embed the logic in the Railtie itself, as you have it now.

So you think we're good to go with the code as-is?

Reply:
Yeah, should be good to go